src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -482,44 +482,50 @@ public static bool TryValidateBlock(BinaryReader reader, ZipArchiveEntry entry)
             }
             else
             {
-                if (reader.BaseStream.Length < reader.BaseStream.Position + extraFieldLength + entry.CompressedLength + 4)
+                if (reader.BaseStream.Length < reader.BaseStream.Position + extraFieldLength + entry.CompressedLength + 12)
                 {
                     return false;
                 }
 
                 reader.BaseStream.Seek(extraFieldLength + entry.CompressedLength, SeekOrigin.Current); // seek to end of compressed file from which Data descriptor starts
-                uint dataDescriptorSignature = reader.ReadUInt32();
-                bool wasDataDescriptorSignatureRead = false;
-                int seekSize = 0;
-                if (dataDescriptorSignature == DataDescriptorSignature)
+                Span<uint> buffer = stackalloc uint[6];
+
+                // let's try to match the smallest possible structure (3 x 4) 32 bit without signature
```

Comment:
Any thoughts on also bringing this comment over as well?

[https://github.com/dotnet/wpf/blob/022c3952a34dad5dd77566768b4558eac182e2b5/src/Microsoft.DotNet.Wpf/src/WindowsBase/MS/Internal/IO/Zip/ZipIOLocalFileDataDescriptor.cs#L45-L58](https://github.com/dotnet/wpf/blob/022c3952a34dad5dd77566768b4558eac182e2b5/src/Microsoft.DotNet.Wpf/src/WindowsBase/MS/Internal/IO/Zip/ZipIOLocalFileDataDescriptor.cs#L45-L58)

I thought it explained the situation pretty well, especially for someone like me who is less familiar with the zip format.